### PR TITLE
Pan os add device group to commands

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3837)

## Description
Add the *device_group* argument to the following commands:
1) `pan-os-list-applications`
2) `pan-os-get-security-profiles`
3) `pan-os-apply-security-profiles`
4) `pan-os-edit-rule`
5) `pan-os-edit-address-group`
6) `pan-os-edit-custom-url-category`
7) `pan-os-edit-edl`
8) `pan-os-edit-service-group`
9) `pan-os-edit-url-filter`

The *device_group* argument for each command should override the *device_group* parameter of the integration.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 1]
```diff
@@ -4230,6 +4233,9 @@ script:
     - description: The tag of the service group to edit.
       isArray: true
       name: tags
+    - description: The device group in which the service group belongs to.
+      name: device_group
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:4237]: why add with a _? Looks like the other args are with a - and the param with a _. Seems like a good way to distinguish. 
I would add DEVICE_GROUP_PARAM_NAME and DEVICE_GROUP_ARG_NAME constants to the code

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 2]
```diff
@@ -7571,8 +7577,8 @@ def initialize_instance(args: Dict[str, str], params: Dict[str, str]):
     # determine a vsys or a device-group
     VSYS = params.get('vsys', '')
 
-    if args and args.get('device-group'):
-        DEVICE_GROUP = args.get('device-group')  # type: ignore[assignment]
+    if args and (device_group := (args.get('device-group') or args.get('device_group'))):
+        DEVICE_GROUP = device_group  # type: ignore[assignment]
     else:
         DEVICE_GROUP = params.get('device_group', None)  # type: ignore[arg-type]
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:7583]:
```suggestion
    DEVICE_GROUP = args and args.get('device-group') or params.get('device_group', None)
```
Replace with constants when you add them in

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml (hunk 1 above)
Comment (GuyAfik) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.yml:4237]: _ was part of the design, ill change it to - to align with the rest

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 3]
```diff
@@ -7570,11 +7578,10 @@ def initialize_instance(args: Dict[str, str], params: Dict[str, str]):
 
     # determine a vsys or a device-group
     VSYS = params.get('vsys', '')
+    DEVICE_GROUP_ARG_NAME = args.get('device-group', '')
+    DEVICE_GROUP_PARAM_NAME = params.get('device_group', '')
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:7582]:
```suggestion
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 4]
```diff
@@ -7570,11 +7578,10 @@ def initialize_instance(args: Dict[str, str], params: Dict[str, str]):
...
 
-    if args and args.get('device-group'):
-        DEVICE_GROUP = args.get('device-group')  # type: ignore[assignment]
-    else:
-        DEVICE_GROUP = params.get('device_group', None)  # type: ignore[arg-type]
+    DEVICE_GROUP = DEVICE_GROUP_ARG_NAME or DEVICE_GROUP_PARAM_NAME
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:7584]:
```suggestion
    DEVICE_GROUP = args.get(DEVICE_GROUP_ARG_NAME) or params.get(DEVICE_GROUP_PARAM_NAME)
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 5]
```diff
@@ -46,6 +46,8 @@
 
 XPATH_SECURITY_RULES = ''
 DEVICE_GROUP = ''
+DEVICE_GROUP_PARAM_NAME = ''
+DEVICE_GROUP_ARG_NAME = ''
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:50]:
```suggestion
DEVICE_GROUP_ARG_NAME = 'device-group'
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 6]
```diff
@@ -46,6 +46,8 @@
 
 XPATH_SECURITY_RULES = ''
 DEVICE_GROUP = ''
+DEVICE_GROUP_PARAM_NAME = ''
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:49]:
```suggestion
DEVICE_GROUP_PARAM_NAME = 'device_group'
```

**Review by JudahSchwartz — APPROVED** [review]
Meant more like this
I think this documents clearer the difference between device-group and device_group.

**Comment by GuyAfik** [comment]
need force merge due to pan-os test-playbooks failing on timeouts

**Pull request merged**